### ISSUE TITLE
all: add OpenCensus tracing

### ISSUE THREAD

**Issue by jba** [google/go-cloud#1126]
We should use the OpenCensus API to add tracing to our APIs. (This proposal covers only tracing, not metrics.)

We should trace the methods of the concrete type that make RPCs. If we trace the concrete type, we only have to write the code once to cover all drivers, and it means new driver authors don't have to do anything.

An exception would be the driver.OpenXXX methods, which should have tracing if they make RPCs.

We should write a small `internal/trace` package to assist. Each traced method will need two lines at the top, to start the trace and defer its end. This is an example from `cloud.google.com/go/datastore`:
```
func (c *Client) Get(ctx context.Context, key *Key, dst interface{}) (err error) {
    ctx = trace.StartSpan(ctx, "cloud.google.com/go/datastore.Get")
    defer func() { trace.EndSpan(ctx, err) }()
```

Note that the error return value is named and passed to the `EndSpan` function so it can be turned into an OpenCensus [Status](https://godoc.org/go.opencensus.io/trace#Status). To do that right, we need to be able to extract an error code, so this proposal is dependent on #1124.

**Comment by vangent** [comment]
See also #629 which overlaps with this. Maybe file a separate issue for metrics and close #629?

**Comment by jba** [comment]
Tracing has been added to all four APIs.
